pkg/controller/incompatibleOperatorController.go

```diff
@@ -0,0 +1,154 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	semver "github.com/blang/semver/v4"
+	"github.com/go-logr/logr"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-olm-operator/pkg/clients"
+	"github.com/openshift/cluster-olm-operator/pkg/utils"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	storage "github.com/operator-framework/helm-operator-plugins/pkg/storage"
+	"github.com/operator-framework/operator-registry/alpha/property"
+	helm "helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type incompatibleOperatorController struct {
+	name                   string
+	nextOCPMinorVersion    *semver.Version
+	kubeclient             kubernetes.Interface
+	clusterExtensionClient *clients.ClusterExtensionClient
+	operatorClient         *clients.OperatorClient
+	logger                 logr.Logger
+}
+
+func NewIncompatibleOperatorController(name string, nextOCPMinorVersion *semver.Version, kubeclient kubernetes.Interface, clusterExtensionClient *clients.ClusterExtensionClient, operatorClient *clients.OperatorClient, eventRecorder events.Recorder) factory.Controller {
+	c := &incompatibleOperatorController{
+		name:                   name,
+		nextOCPMinorVersion:    nextOCPMinorVersion,
+		kubeclient:             kubeclient,
+		clusterExtensionClient: clusterExtensionClient,
+		operatorClient:         operatorClient,
+		logger:                 klog.FromContext(context.TODO()).WithName(name),
```

Comment:
Do we have a logger we could pass in rather than constructing from `klog.FromContext`? It feels a little weird to call that from a function what will never expect a context.

If not, maybe there's a different `klog` function that could just hand us a new logger directly rather than referencing through a context that we know won't already have one?